### PR TITLE
fix: invalid handling of OONI Run v1 link with partial URL

### PR DESCRIPTION
**Pull request by aanorbel**
Fixes  https://github.com/ooni/probe/issues/2515

## Proposed Changes

  - Modified `OoniRunViewController#validateLink` to use updated code that allows a better validation.

Simulator Screenshot - iPhone 14 Pro - 2023-07-28 at 08 40 57

### CODE REVIEW

File: ooniprobe/View/OONIRun/OoniRunViewController.m [hunk 1]
```diff
@@ -190,16 +190,41 @@ -(void)validateAndAddURLs{
...
+    // Check if the host property of the NSURLComponents object is nil; if so, the URL is not valid
+    if (components.host == nil) {
+        return NO;
+    }
+    // Convert the NSURLComponents object to an NSURL object
+    NSURL *URL = components.URL;
+    // Check if the NSURL object is nil, indicating that the URL is not valid
+    if (URL == nil) {
+        return NO;
+    }
```
Comment (bassosimone) [inline ooniprobe/View/OONIRun/OoniRunViewController.m:225]:
```suggestion
```

We don't need this part.

**Review by bassosimone — CHANGES_REQUESTED** [review]
Please, write tests covering each of the possible error conditions. You can use https://www.onlinegdb.com/V81JCf11o to get some inspiration about possible test cases.

File: ooniprobe/View/OONIRun/OoniRunViewController.m [hunk 2]
```diff
@@ -190,16 +190,35 @@ -(void)validateAndAddURLs{
...
+ * @return YES if the input string is a valid URL; otherwise, NO.
  */
-- (BOOL)validateLink:(NSString *)link
-{
-    NSString *regex = @"(?i)(http|https)(:\\/\\/)([^ .]+)(\\.)([^ \n]+)";
-    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"SELF MATCHES %@", regex];
-    return [predicate evaluateWithObject:link];
+- (BOOL)validateLink:(NSString *)urlString {
+    // Check if the length of the input string is greater than 2083 characters; if so, the URL is not valid
+    if (urlString.length > 2083) {
```
Comment (hellais) [inline ooniprobe/View/OONIRun/OoniRunViewController.m:201]: I would suggest removing this constraint on the maximum URL length. AFAIK this is a limitation that only used to exist in older internet explorer versions, but it's not really something part of the specification for URLs.

**Review by hellais — COMMENTED** [review]
Suggested a small fix, but otherwise LGTM 🐳

**Comment by hellais** [comment]
Lgtm

**Pull request merged**